File: app/src/main/java/ch/bailu/aat/activities/AbsKeepScreenOnActivity.java
package ch.bailu.aat.activities;

import android.os.Bundle;

import ch.bailu.aat.gpx.InfoID;
import ch.bailu.aat.util.ui.Backlight;

public abstract class AbsKeepScreenOnActivity extends ActivityContext {

    private Backlight backlight;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        backlight = new Backlight(getWindow(), getServiceContext());

        addTargets(backlight, InfoID.TRACKER);
    }

    @Override
    public void onDestroy() {
        backlight.close();
        super.onDestroy();
    }

}

File: aat-android/src/main/kotlin/ch/bailu/aat/activities/AbsKeepScreenOnActivity.kt
package ch.bailu.aat.activities

import android.os.Bundle
import ch.bailu.aat.util.ui.Backlight
import ch.bailu.aat_lib.gpx.information.InfoID

abstract class AbsKeepScreenOnActivity : ActivityContext() {
    private var backlight: Backlight? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        backlight = Backlight(this, serviceContext).apply {
            dispatcher.addTarget(this, InfoID.TRACKER)
        }
    }

    override fun onResumeWithService() {
        super.onResumeWithService()
        backlight?.setBacklightAndPreset()
    }

    override fun onDestroy() {
        backlight?.close()
        backlight = null
        super.onDestroy()
    }
}
